plugins/database/postgresql/postgresql.go

```diff
@@ -501,3 +516,48 @@ func (p *PostgreSQL) RotateRootCredentials(ctx context.Context, statements []str
 	p.RawConfig["password"] = password
 	return p.RawConfig, nil
 }
+
+// containsMultilineStatement is a best effort to determine whether
+// a particular statement is multiline, and therefore should not be
+// split upon semicolons. If it's unsure, it defaults to false.
+func containsMultilineStatement(stmt string) bool {
+	// We're going to look for the word "END", but first let's ignore
+	// anything the user provided within single or double quotes since
+	// we're looking for an "END" within the Postgres syntax.
+	literals, err := extractQuotedStrings(stmt)
+	if err != nil {
+		return false
+	}
+	stmtWithoutLiterals := ""
+	for _, literal := range literals {
+		stmtWithoutLiterals = strings.Replace(stmt, literal, "", -1)
+	}
+	// Now look for the word "END" specifically. This will miss any
+	// representations of END that aren't surrounded by spaces, but
+	// it should be easy to deal with. This approach was chosen to
+	// avoid accidently matching with unintended words like "APPEND".
+	return strings.Contains(stmtWithoutLiterals, " END ")
+}
+
+// extractQuotedStrings extracts 0 or many substrings
+// that have been single- or double-quoted. Ex:
+// `"Hello", silly 'elephant' from the "zoo".`
+// returns [ `"Hello"`, `'elephant'`, `"zoo"` ]
+func extractQuotedStrings(s string) ([]string, error) {
+	var result []string
+
+	// Find all double-quoted sub-strings.
+	reg, err := regexp.Compile(`"(.*?)"`)
```

Comment:
Why not include the delimiters inside the parens so they count as part of the match?  That way you won't inadvertently mangle the statement if it contains something like `"role"`.

Reply:
Sure, addressed in https://github.com/hashicorp/vault/pull/8512/commits/59979df86837a536a29dc470d8b5e6f62a0a475b